### PR TITLE
[reconfiguration] support reconfiguration triggered by validator key …

### PR DESCRIPTION
**Pull request by sblackshear**
…rotation

- Invoke ValidatorSet.reconfigure() from Block.prologue()
- Add logic to ValidatorSet.reconfigure() that looks at each ValidatorConfig resource belonging to a validator, determines if key rotation has occurred, and copies over the new key if so
- Add transaction script for validator key rotation
- Add tests that perform a validator key rotation + check that reconfiguration occurs, and perform a duplicate (i.e., same-key) rotation + check that no reconfiguration occurs

## Motivation

More progress on https://github.com/libra/libra/issues/1382

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

New E2E tests.

### CODE REVIEW

File: language/stdlib/modules/validator_set.mvir [hunk 1]
```diff
@@ -74,53 +74,108 @@ module ValidatorSet {
...
             }
         );
 
         return;
     }
 
+   // Return true if the ValidatorInfo given as input is different than the ValidatorConfig
+    // published at validator_info.addr. Aborts if there is no ValidatorConfig at
+    // validator_info.addr
+    public validator_info_changed(validator_info: &mut Self.ValidatorInfo): bool {
```
Comment (zekun000) [inline language/stdlib/modules/validator_set.mvir]: the function name is a little confusing to me, it not only tests the change but also applies it

Comment (sblackshear) [reply language/stdlib/modules/validator_set.mvir]: Will change the name.
